### PR TITLE
use pts_lat and pts_lon for single point cases

### PR DESCRIPTION
**Pull request by jedwards4b**
### Description of changes
Instead of defining a domain file with a single lat lon point we use the PTS_LAT and PTS_LON variables to input
the location to be used to create a mesh.   This requires the cime changes in https://github.com/ESMCI/cime/pull/3868

### Specific notes

Contributors other than yourself, if any: 

CMEPS Issues Fixed (include github issue #):

Are changes expected to change answers?
 - [X] bit for bit
 - [ ] different at roundoff level
 - [ ] more substantial 

Any User Interface Changes (namelist or namelist defaults changes)?
 - [ ] Yes
 - [ ] No

Testing performed if application target is CESM:(either UFS-S2S or CESM testing is required):
- [ ] (recommended) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines:
   - details (e.g. failed tests):
- [ ] (recommended) CESM testlist_drv.xml
   - machines and compilers:
   - details (e.g. failed tests):
- [ ] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):
- [ ] (other) please described in detail
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-coupled:
- [ ] (recommended) UFS-coupled testing
   - description:
   - details (e.g. failed tests):

Testing performed if application target is UFS-HAFS:
- [ ] (recommended) UFS-HAFS testing
   - description:
   - details (e.g. failed tests):

Hashes used for testing:
- [ ] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch: nuopc_dev
  - hash:
- [ ] UFS-coupled, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:
- [ ] UFS-HAFS, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:

### CODE REVIEW

**Review by mvertens — CHANGES_REQUESTED** [review]
Don't we need to set single_column accordingly as well here?
I thought the logic was that if single_column was was true we needed to use a domain file in order to do a nearest neighbor search - single_column_domainfile.
So how are we setting PTS_MODE to enforce that? This would be in config_component.xml.
Also - config_component.xml needs to be changed to rewrite the following comments for PTS_LON and PTS_LAT
`Longitude to find nearest points for points mode (only used if PTS_MODE is TRUE)`
` Latitude to find nearest points for points mode (only used if PTS_MODE is `TRUE)`
I am actually wondering if we want to get rid of points mode and JUST use the single_column_domainfile. If that is set then you will do a nearest neighbor search using PTS_LON and PTS_LAT otherwise you just use them. It might be easier than having both that filename AND the PTS_MODE to be set. Thoughts?

**Comment by ekluzek** [comment]
> Don't we need to set single_column accordingly as well here?
> I thought the logic was that if single_column was was true we needed to use a domain file in order to do a nearest neighbor search - single_column_domainfile.
> So how are we setting PTS_MODE to enforce that? This would be in config_component.xml.
> Also - config_component.xml needs to be changed to rewrite the following comments for PTS_LON and PTS_LAT
> `Longitude to find nearest points for points mode (only used if PTS_MODE is TRUE)`
> `Latitude to find nearest points for points mode (only used if PTS_MODE is`TRUE)`
> I am actually wondering if we want to get rid of points mode and JUST use the single_column_domainfile. If that is set then you will do a nearest neighbor search using PTS_LON and PTS_LAT otherwise you just use them. It might be easier than having both that filename AND the PTS_MODE to be set. Thoughts?

Yes it seems to me we need to distinguish between running in single_column mode and running for a single point. single_column mode expects global datasets and then runs on the gridcell that's closest to the PTS_LAT,PTS_LON.

For single_column mode running with SCAM I think you DO want to use the global domain file because SCAM can bring in other information from the surrounding grid cells. For CTSM running on a single point the size, mask and area of the grid cell doesn't matter, but I'm not so sure of that for SCAM inside of CAM. They do have a mode of using files that bring in lateral advection from surrounding grid cells.

But, one thing that's critically important is that you you want single_column to be FALSE when running for a single point grid. You don't want to exercise the extra logic in finding the closest grid cells, especially because in CTSM that mode can NOT currently restart.

We probably need to bring in some SCAM people here.

File: cime_config/buildnml [hunk 1]
```diff
@@ -43,7 +43,8 @@ def _create_drv_namelists(case, infile, confdir, nmlgen, files):
     config['MPILIB'] = case.get_value('MPILIB')
     config['OS'] = case.get_value('OS')
     config['glc_nec'] = 0 if case.get_value('GLC_NEC') == 0 else case.get_value('GLC_NEC')
-    config['single_column'] = 'true' if case.get_value('PTS_MODE') else 'false'
+    if case.get_value("ATM_NX") == 1 and case.get_value("ATM_NY") == 1 and case.get_value("ATM_DOMAIN_FILE") != "UNSET":
+        config['single_column'] = 'true'
```
Comment (ekluzek) [inline cime_config/buildnml:47]: We only want single_column = True for the case when you are running SCAM mode with global datasets. Turning single_column to true for cases of a single point exercises different code, that you don't need to. And it has limitations like CTSM can't restart. So we can't set it up this way. I really think that PTS_MODE needs to come back in. I like the idea of simplifying when we can, but I don't see that we can do that here.

**Comment by jedwards4b** [comment]
@ekluzek I have tested both cases (scam and single point) successfully.  If you can produce a counter example I would be happy to consider it.

**Comment by jedwards4b** [comment]
Superseded by #164

**Pull request closed without merging**